### PR TITLE
Add Telegram notification provider

### LINKED ISSUE #132: Integrate telegram
**Is your feature request related to a problem? Please describe.**
We already have the notifications system base, but providers are not integrated, yet.

**Describe the solution you'd like**
Start with telegram using telegram-bot or anything else.

**Describe alternatives you've considered**
No

**Additional context**
Np

### PR DESCRIPTION
**Pull request by XaviTorello**
## Description

It provides the Telegram notification provider that interacts with a Telegram Bot to notify to the configured ChatIDs.

When a new job is submitted, this provider notifies to all configured chats via Telegram.

image

Fix #132 

### Considerations

1) To avoid implementing Django Sites, the new var `BASE_URL` has been provided (by default pointing to `http://localhost:8080`)

2) `@anemperfeinabot` has been created and should be defined as Administrator in the destination groups

### New env vars
```
# Our base URL, needed for the links generation
BASE_URL=https://nemperfeina.cat

# Telegram provider should be enabled
NOTIF_TELEGRAM_ENABLED=True  

# Telegram Bot Token
TELEGRAM_TOKEN=XXXXXXX  
# Comma-separated list of chatIDs (can be @XXX or the ID number), ie `@pygrn`
TELEGRAM_CHAT_IDS=@groupName,@channelName,@chatID
```

### Test it

Currently has been deployed to https://testing.nemperfeina.cat

- 1) Login
- 2) Create a new job offer
- 3) Reach notification into test group

### CODE REVIEW

File: jobs/settings.py [hunk 1]
```diff
@@ -10,6 +10,7 @@
 
 env = environ.Env()
 
+BASE_URL = env("BASE_URL", default="http://localhost:8000")
```
Comment (jbagot) [inline jobs/settings.py:13]: localhost? Or app?

Comment (XaviTorello) [reply jobs/settings.py:13]: Use localhost will support both developments (local and Docker)

// Base composition exposes `tcp#8000`, so will also work

Anyway, if needed can be overrided using `.env`

File: notifications/providers/telegram.py [hunk 2]
```diff
@@ -1,8 +1,26 @@
...
 
 class Telegram(ProviderBase):
     code = "telegram"
 
     def _publish(self, tpl: str):
-        pass
+        """
+        Publish notifications to Telegram using a BOT
+        """
+        settings = PROVIDERS_SETTINGS.get(self.code)
```
Comment (francescarpi) [inline notifications/providers/telegram.py:14]: Maybe would be better to set a method in the parent class (ProviderBase) name `get_settings` and move this functionality there.

**Review by francescarpi — COMMENTED** [review]
Hi bro! Excellent work!
I only posted a little improvement.

File: notifications/providers/telegram.py (hunk 2 above)
Comment (XaviTorello) [reply notifications/providers/telegram.py]: Done!

**Review by francescarpi — APPROVED** [review]
Good job!

**Comment by francescarpi** [comment]
All right for me. You can merge it if you want.

**Review by ytturi — APPROVED** [review]
Nice 👍

**Pull request merged**